rpc/query_server.go

```diff
@@ -782,8 +782,12 @@ func (s *QueryServer) EthGetTransactionReceipt(hash eth.Data) (*eth.JsonTxReceip
 	txResults, err := s.BlockStore.GetTxResult(blockResult.Block.Data.Txs[txReceipt.TransactionIndex].Hash())
 	if err != nil {
 		if strings.Contains(errors.Cause(err).Error(), "not found") {
-			// return nil response if cannot find hash
-			return nil, nil
+			txResults, err := s.BlockStore.GetTxResultByHeightAndIndex(&height, int(txReceipt.TransactionIndex))
+			if err != nil {
+				// return receipts without blockresult info
+				return completeReceipt(nil, blockResult, &txReceipt), nil
```

Comment:
I think we should return nil in this case, like we used to. Creating a receipt with an inaccurate status seems worse than just returning nothing.

Reply:
Done